backend/src/test/java/ai/verta/modeldb/GlobalSharingTest.java

```diff
@@ -0,0 +1,451 @@
+              orgResourceId = createProjectResponse.getProject().getId();
+              break;
+            case DATASET:
+              // Create dataset
+              CreateDataset createDatasetRequest =
+                  DatasetTest.getDatasetRequest("dataset-" + new Date().getTime());
+              createDatasetRequest =
+                  createDatasetRequest
+                      .toBuilder()
+                      .setWorkspaceName(organization.getName())
+                      .setVisibility(resourceVisibility)
+                      .setCustomPermission(customPermission)
+                      .build();
+              CreateDataset.Response createDatasetResponse =
+                  datasetServiceStub.createDataset(createDatasetRequest);
+              orgResourceId = createDatasetResponse.getDataset().getId();
+              break;
+            case REPOSITORY:
+            default:
+              // Create repository
+              String repoName = "repository-" + new Date().getTime();
+              SetRepository.Response createRepositoryResponse =
+                  repositoryServiceStub.createRepository(
+                      SetRepository.newBuilder()
+                          .setId(
+                              RepositoryIdentification.newBuilder()
+                                  .setNamedId(
+                                      RepositoryNamedIdentification.newBuilder()
+                                          .setName(repoName)
+                                          .setWorkspaceName(orgName)))
+                          .setRepository(
+                              Repository.newBuilder()
+                                  .setVisibility(resourceVisibility)
+                                  .setName(repoName)
+                                  .setCustomPermission(customPermission))
+                          .build());
+              repository = createRepositoryResponse.getRepository();
+              orgResourceId = String.valueOf(repository.getId());
+              break;
+          }
+        } else {
+          collaboratorServiceStub.setResources(
+              SetResources.newBuilder()
+                  .setWorkspaceName(organization.getName())
+                  .setResources(
+                      Resources.newBuilder()
+                          .addResourceIds(orgResourceId)
+                          .setResourceType(
+                              ResourceType.newBuilder()
+                                  .setModeldbServiceResourceType(
+                                      resourceType
+                                              == ModelDBResourceEnum.ModelDBServiceResourceTypes
+                                                  .DATASET
+                                          ? ModelDBResourceEnum.ModelDBServiceResourceTypes
+                                              .REPOSITORY
+                                          : resourceType)
```

Comment:
We seem to be ignoring the type `DATASET`. I'm not sure why. Let's make sure everything except for the internals of modeldb treat dataset and repository as two completely independent types. The fact that datasets are implemented as repositories is an internal implementation detail and should not be exposed outside of the internal logic.

Reply:
Internally dataset is repository so all bindings and resources are created in UAC for REPOSITORY. I'm not sure how to use DATASET here. When in Webapp you open a dataset, there is collaborator service used for REPOSITORY so bindings are created for REPOSITORY.

Reply:
And MDB don't check any DATASET bindings. What should be done to do as you suggest? All UAC endpoints for Dataset should start working with repositories when called? Can it be done?

Reply:
>all bindings and resources are created in UAC for REPOSITORY

This is very surprising to me. I'd say this is definitely a bug, but not one that we need to resolve right now. We should have UAC consider repositories and datasets as completely separate entities.

Reply:
this is the case which i have asked when we have stated working on visibility @conradoverta Slack thread: https://vertaai.slack.com/archives/CJG0NL86B/p1608212405074500